src/hooks/useApproveAndSwap.ts

```diff
@@ -169,18 +181,27 @@ export function useApproveAndSwap(): (
           Math.round(new Date().getTime() / 1000 + 60 * deadline),
           txnArgs,
         ] as const
+        console.debug("swap - direct", args)
         swapTransaction = await (state.swapContract as NonNullable<
           typeof state.swapContract // we already check for nonnull above
         >).swap(...args)
-        console.debug("swap - direct", args)
       } else if (state.swapType === SWAP_TYPES.SYNTH_TO_SYNTH) {
-        // TODO: support synth to synth
-        throw new Error("Synth to Synth swaps not yet supported")
+        const args = [
+          utils.formatBytes32String(state.from.symbol),
+          state.from.amount,
+          utils.formatBytes32String(state.to.symbol),
+          account,
+          SYNTH_TRACKING_ID,
+        ] as const
+        console.debug("swap - synthToSynth", args)
+        swapTransaction = await baseSynthetixContract?.exchangeWithTracking(
+          ...args,
+        )
       } else {
         throw new Error("Invalid Swap Type, or contract not loaded")
       }
 
-      notifyHandler(swapTransaction.hash, "swap")
+      notifyHandler(swapTransaction?.hash || "", "swap")
```

Comment:
should we be calling `notifyHandler` with an empty tx hash?  wouldn't it be preferable to do something like:
```js
if (swapTransaction?.hash) {
  notifyHandler(swapTransaction?.hash, "swap")
}
```